### PR TITLE
Add `device` key into the `hardware` specification

### PR DESCRIPTION
**Pull request by juk**

### CODE REVIEW

**Comment by lkotek** [comment]
Just an idea that crossed my mind that we can maybe take into consideration, because this PR will add new very generic item, but there are already some existing hardware specifications like `cpu`, `disk`, `network` etc. 

1. Should we consider these just like some special cases of `device`? 
2. Should the `disk` (for example) be just a wrapper (or syntax sugar) for `device` in the future?
3. Should the `network` (for example) inherit its keys from `device` and add some specific ones for `network`? 
4. Or should we keep and treat existing hardware "devices" and new `device` from this PR completely separately?

**Comment by happz** [comment]
Probably a good topic for https://github.com/teemtee/tmt/discussions/1643 before moving further with this particular PR, I find @lkotek's comments worth a deeper discussion.

**Comment by thrix** [comment]
@lkotek and others, so we agreed for now to:

1. treat the `devices` as separate configuration entities
2. accept for now the fact there might be overlaps with `network`, `disk`, `gpu` and similar
3. revisit if issues arise later

File: tmt/schemas/provision/hardware.yaml [hunk 1]
```diff
@@ -97,6 +97,37 @@ definitions:
...
+      device-id:
+        type: string
+
+      driver:
+        type: string
+
+      vendor-name:
+        type: string
+
+      vendor-id:
```
Comment (happz) [inline tmt/schemas/provision/hardware.yaml:117]: `vendor-id` is not documented, and I suppose it's a "typo" and the proper name should be just `vendor`.

Comment (happz) [reply tmt/schemas/provision/hardware.yaml:117]: The same seems to apply to `device-id`.

Comment (happz) [reply tmt/schemas/provision/hardware.yaml:117]: And both `vendor` and `device` should allow both strings and integers, some users might forget about quotes, for example.

```yaml
        anyOf:
          - type: string
          - type: integer
```

Comment (juk) [reply tmt/schemas/provision/hardware.yaml]: Thank you for @happz! Update PR

**Review by happz — APPROVED** [review]

**Review by psss — APPROVED** [review]
Looks good, thanks!

**Pull request merged**